.github/workflows/geth-with-protocol.yml

```diff
@@ -5,23 +5,26 @@ jobs:
   docker:
     runs-on: ubuntu-20.04
     steps:
-      - name: docker login
-        env:
-          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
-          DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
-        run: |
-          # docker login
-          docker version
-          docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
 
       - name: Check out code
         uses: actions/checkout@v1
 
-      - name: docker build
-        run: |
-          cd geth-with-protocol
-          docker build -t livepeer/geth-with-livepeer-protocol:confluence .
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
 
-      - name: docker push
-        run: |
-          docker push livepeer/geth-with-livepeer-protocol:confluence
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          context: ./geth-with-protocol
+          platforms: linux/arm64, linux/amd64
```

Comment:
Do you mean from a best practices perspective or that it's just plain wrong? Because i tested it on a fork and it works - also, my yaml linter didn't complain